sorting/topological_sort.cpp

```diff
@@ -0,0 +1,98 @@
+/**
+ * @file
+ * @brief Topological sort algorithm
+ *
+ * Implementation Details -
+ *      Topological sorting for Directed Acyclic Graph (DAG) is a linear ordering
+ *	of vertices such that for every directed edge uv, vertex u comes before v in the
+ *	ordering. Topological Sorting for a graph is not possible if the graph is not a DAG.
+ */
+#include <algorithm>
+#include <iostream>
+#define MAX 10
+int n, adj[MAX][MAX];
+int front = -1, rear = -1, Queue[MAX];
+void create_graph() {
+	int i, max_edge, origin, dest;
+	std::cout<<"Enter number of Vertex\n";
+	std::cin>>n;
+	max_edge = n*(n-1);

```

Comment:
it will be a more flexible approach if along with vertices you take number of edges input from user as well.